homeassistant/loader.py

```diff
@@ -54,10 +54,37 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+#
+# Integration.get_component will check preload platforms and
+# try to import the code to avoid a thundering heard of import
+# executor jobs later in the startup process.
+#
+# default platforms are prepopulated in this list to ensure that
+# by the time the component is loaded, we check if the platform is
+# available.
+#
+# This list can be extended by calling async_register_preload_platform
+#
+BASE_PRELOAD_PLATFORMS = [
+    "config",
+    "diagnostics",
+    "energy",
+    "group",
+    "logbook",
+    "hardware",
+    "media_source",
+    "recorder",
+    "repairs",
+    "system_health",
+    "trigger",
+]
```

Comment:
Maybe we make this list a new [generated](https://github.com/home-assistant/core/tree/dev/homeassistant/generated) file that hassfest can keep up to date from the manifest. That bootstrap can call `async_register_preload_platform` for all the ones that we know are going to be setup so if its not being used we don't add it to this list.... maybe thats a bit overkill

Reply:
I think we can add that later if we need it as it might be a case YAGNI and every new one of these we add makes startup slower so I'm not super inclined to increase this list so much without a redesign